### PR TITLE
Update superlu to 5.2.1 + fix stdint.h issue on MSVC

### PR DESCRIPTION
**Pull request by pv**
Update SuperLU to version 5.2.1 (it has only very minor changes).

Use stdint.h from msinttypes for Microsoft compilers.

### CODE REVIEW

**Comment by rgommers** [comment]
Tests all finished and OK, just a timeout right at the end.

**Comment by rgommers** [comment]
Build changes LGTM.

File: scipy/sparse/linalg/dsolve/SuperLU/README [hunk 1]
```diff
@@ -1,36 +1,5 @@
...
-PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
-CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
-EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
-PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
-PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
-  
+		SuperLU (Version 5.2.x)
```
Comment (matthew-brett) [inline scipy/sparse/linalg/dsolve/SuperLU/README]: Should this be `5.2.1`?

**Review by matthew-brett — APPROVED** [review]
Thanks - looks right to me.

File: scipy/sparse/linalg/dsolve/SuperLU/License.txt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+
+
+-----------------
+External software
+-----------------
+In the ILU preconditioner driver routine xGSISX, we use MC64 routine
+to pre-permute large entries to the main diagonal. This use path can
+be removed for commercial users. MC64 routine carries the following notice:
```
Comment (WarrenWeckesser) [inline scipy/sparse/linalg/dsolve/SuperLU/License.txt]: Are we using the "MC64 routine"?  (Sorry, I haven't taken a close look at the code.)

IANAL, but this license looks far more restrictive than the BSD or MIT licenses.  The following seem to place additional burdens on downstream users, beyond what we expect from the BSD or MIT license:

* The user shall use reasonable endeavours to notify the authors of the package of this publication.
* The user shall make available free of charge to the authors for any purpose all information relating to any alteration or addition made to this package for the purposes of extending the capabilities or enhancing the performance of this package.
* The user shall not pass this code directly to a third party without the express prior consent of the authors.

If we are not using the MC64 routine, then can we remove this part of the license?

**Comment by pv** [comment]
No, we're not using mc64. The source code for it has not been checked in.

File: scipy/sparse/linalg/dsolve/SuperLU/License.txt (hunk 2 above)
Comment (rgommers) [inline scipy/sparse/linalg/dsolve/SuperLU/License.txt]: > @pv: No, we're not using mc64. The source code for it has not been checked in.

Probably worth removing that part of the license then as @WarrenWeckesser suggests.

**Comment by pv** [comment]
Removed irrelevant license text.

File: scipy/sparse/linalg/dsolve/SuperLU/README (hunk 1 above)
Comment (pv) [inline scipy/sparse/linalg/dsolve/SuperLU/README]: Maybe, but it's like that in the source release...

**Comment by rgommers** [comment]
In it goes, thanks Pauli!

**Pull request merged**